homeassistant/components/lektrico/coordinator.py

```diff
@@ -0,0 +1,56 @@
+"""Coordinator for the Lektrico Charging Station integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from aiohttp import ClientSession
+from lektricowifi import DeviceConnectionError, lektricowifi
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+
+class LektricoDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """The device class for Lektrico device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        friendly_name: str,
+        host: str,
+        session: ClientSession,
+    ) -> None:
+        """Initialize a Lektrico Device."""
+        self.device = lektricowifi.Device(
+            host,
+            session=session,
+        )
+        self.friendly_name = friendly_name.replace(" ", "_")
+        self._name = friendly_name
+        self._update_fail_count = 0
+        self._info = None
+        super().__init__(
+            hass, LOGGER, name=f"{DOMAIN}-{self._name}", update_interval=SCAN_INTERVAL
```

Comment:
I don't think adding DOMAIN to the name adds value, any prints from the `DataUpdateCoordinator` will already be attribute to `lektrico` becuase we pass it our logger.

Reply:
I removed it. Thanks.